src/app/services/ws.service.ts

```diff
@@ -221,18 +209,25 @@ export class WebSocketService {
     });
   }
 
-  sub<T = any>(name: string): Observable<T> {
-    const nom = name.replace('.', '_'); // Avoid weird behavior
+  /**
+   * This method subscribes to the provided api end point for real time updates
+   * @param api The api end point to subscribe to
+   * @param subscriptionId The unique id that will be used as request id and can be
+   * used to unsubscribe from the websocket subscription with the `unsub(api, subscriptionId)`
+   * method
+   * @returns
+   */
+  sub<T = any>(api: string, subscriptionId?: string): Observable<T> {
```

Comment:
The purpose of the `subscriptonId` here is to provide a way for the calling component/service to unsubscribe from these subscriptions. This is a necessary step because if you don't unsubscribe from the subscription and leave it forever open, you can't subscribe to it again later on anywhere else if you have. The middleware won't respond to the new subscription request.

Reply:
Here, by "unsubscribe" I don't mean unsubscribing from an observable. I mean calling `WebSocketService.send` method with `subscriptionId` and `unsub` message like so `WebSocketService.send({ id: subscriptionId, msg: 'unsub'})`.